src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -314,6 +314,61 @@ private void RemoveEntry(NodeSearchElement entry)
                     parent.Items.RemoveAt(0);
                 target = parent;
             }
+
+            // After removal of category "target" can become the class.
+            // In this case we need to add target to existing classes contaiiner 
+            // (ClassesNodeCategoryViewModel) or create new one.
+            // For example we have a structure.
+            //
+            //                         Top
+            //                          │
+            //                       Sub1_1  
+            //             ┌────────────┤       
+            //          Sub2_1       Classes 
+            //    ┌────────┤            │     
+            // Classes     Member2   Sub2_2   
+            //    │                     │     
+            // Sub3_1                   Member3
+            //    │                            
+            //    Member1   
+            // 
+            // Let's remove "Member1". Before next code we have removed entry "Member1" and
+            // categories "Sub3_1", "Classes". "Sub2_2" is "target" as soon as it has one item in
```

Comment:
I think you meant `Sub2_1` is the `target` as soon as it has only one item in collection?